packages/scripts/src/upgrade-tests/voting2/migrationUtils.ts

```diff
@@ -165,3 +178,47 @@ export const isContractInstance = async (address: string, functionSignature: str
   const encodedSignature = hre.web3.eth.abi.encodeFunctionSignature(functionSignature).slice(2);
   return code.includes(encodedSignature);
 };
+
+export const isGovernorV2Instance = async (address: string): Promise<boolean> => {
+  return isContractInstance(address, "propose((address,uint256,bytes)[],bytes)");
+};
+
+export const isProposerV1Instance = async (address: string): Promise<boolean> => {
+  return isContractInstance(address, "propose((address,uint256,bytes)[])");
+};
+
+export const isVotingV2Instance = async (address: string): Promise<boolean> => {
+  return await isContractInstance(address, "stake(uint256)");
+};
+
+export const proposeEmergency = async (
+  emergencyProposerAddress: string,
+  votingToken: VotingTokenEthers,
+  adminProposalTransactions: AdminProposalTransaction[],
+  afterProposeCallback?: () => void
+): Promise<void> => {
+  const signer = await hre.ethers.getSigner(FOUNDATION_WALLET);
+
+  const emergencyProposer = await getContractInstance<EmergencyProposerEthers>(
+    "EmergencyProposer",
+    emergencyProposerAddress
+  );
+  const emergencyProposeBond = await emergencyProposer.quorum();
+  const allowance = await votingToken.allowance(await signer.getAddress(), emergencyProposer.address);
+  if (allowance.lt(emergencyProposeBond)) {
+    console.log("Approving emergency proposer bond");
+    const approveTx = await votingToken.connect(signer).approve(emergencyProposer.address, emergencyProposeBond);
+    await approveTx.wait();
+  }
+  console.log("SENDING PROPOSAL TXS TO EMERGENCY PROPOSER");
+  await emergencyProposer.connect(signer).emergencyPropose(adminProposalTransactions);
+  console.log("Proposal done!🎉");
+  console.log("\n❓ OPTIONAL: Simulate and execute the emergency proposal with the following command: ");
+  console.log(
+    `
+  ${NEW_CONTRACTS.emergencyProposer}=${process.env[NEW_CONTRACTS.emergencyProposer]} \\
+  ${EMERGENCY_EXECUTOR}=${process.env[EMERGENCY_EXECUTOR]} \\
+  yarn hardhat run ./src/admin-proposals/simulateEmergencyProposal.ts --network ${hre.network.name}`.replace(/  +/g, "")
```

Comment:
I feel like I'm missing something. The way I'm reading this is that it would strip all whitespace in this entire log message. is that right? Was that intentional?

Reply:
I forgot to replace that one with the [formatIndentation](https://github.com/UMAprotocol/protocol/blob/a18f5bc0ce1bd50106738def5487c69b6a376e75/packages/scripts/src/upgrade-tests/voting2/migrationUtils.ts#L59) function. The treatment removes the extra indentation when console logging multiline strings. It's convenient to make the commands we log readable in the code and log them without extra indentation in the console.